src/python/database.py

```
@@ -288,25 +320,26 @@ def link_task_label(self, task_id, label_id):
         with sqlite3.connect(self.db_file) as conn:
             cursor = conn.cursor()
             cursor.execute(query, (task_id, label_id))
+            log.info("database.py - Linked task %s to label %s", task_id, label_id)
     
     def clear_task_labels(self, task_id):
         query = "DELETE FROM task_labels WHERE task_id = ?"
-        with self.conn:
+        with sqlite3.connect(self.db_file) as conn:
```

Ruff (0.7.0)

Local variable `conn` is assigned to but never used

Remove assignment to unused variable `conn`

(F841)